### ISSUE TITLE
Cooldowntracking Aura for Shadowfiend for Disciplinepriest won't load.

### ISSUE THREAD

**Issue by Haxxen** [WeakAuras/WeakAuras2#1938]
## Describe the bug
I wanted to track the Cooldown of shadowfiend. This is a spell shared for Discipline and Shadowpriest. The same Aura wont work for Disciplin but works normal for shadow. It just wont do anything in Discipline-Spec. Even if i use a built in template to make a Aura for cooldowntracking for Shadowfiend, it wont do anything in Discipline. 

Shadowfiend duration-tracking via Totem-Tracker works.

All my other CD-Tracking WAs are Setup per Duplicate and work. Even if you take the ShadowfiendWA and write another Spell Ability name in the trigger it will work as usual.

**Do you have an error log of what happened?**
No.

### To Reproduce

Get on a Discipline Priest --> Make a new Aura: Use Template -> Symbol standard -> Abilities -> Shadowfiend --> Cooldowntracking ---> Close Weakaurawindow --> Aura won't show/load (doesnt matter if Spell is on Cooldown or ready, or how the Spell-CD Trigger is set.)

I can post the string if needed, but i think because its a built in Template u wont need it. (im no programmer, so pls just say if it is needed)

Could only test on a Maghar Orc female Priest. Dont have a second one.

 

**Did you try having WeakAuras as the only enabled addon and everything else (especially something like ElvUI) disabled?**

Yes. Did it with only WA enabled.

**Which version of WeakAuras are you using?**

2.16.1

**Are you on World of Warcraft Classic or Retail?**

Retail

**Was it working in a previous version? If yes, which was the last good one?**

It was working. (logged in and had a gap in my interface) Didn't play the Priest for a bit. last time at the beginnging of BFA. Dont know when or which version.

**Comment by emptyrivers** [comment]
I followed the steps to reproduce, and the resulting shadowfiend aura behaves correctly:

```
!LsvxVTTnu0FndO7Hoyh7g08OJRYQbCL8evkYEH20susCrMuLKsoopKF79EjP8xdOladqqI6Ylp3dp3pOJPzuIGgpEYTJV7okXqjFH3tj7OKc8b29LKYsd3sJhrjhoBntMxR0RucPLsMhfNfLsj5kvtHAVKSx0YP3FYqurfNUfIvUsI2TArvfxBGie7irWaU00TL3ZL2vAEP4fkHSkA5sWJdaKeJLz7aA2jfqCBByhGdb)5obWQoRciENMzfqKiJH9m81vCjxlYj1Q9jU4FPbIb(oRzp7aGSKTJJ8cqtZznKwEttmydisndUlLcUSWdR54EaKUFwueo6WLG0v6UeRNpJKTMKnlnl4Q7CXtMoDYekjq)5b56TnR0Qkn3yEBZhCm439r0Qz5pJrRGVfqoZPjFnA5QhECjg2o5jH0Igy5wrppZB8BQciKFC8iChiXX1swZ3H0aQvX3ClL8mN3odOxUnfvqmLXKIDE1mZj)AlUyqIxhslCiXc39qwsQKCmg7yc5737sHuyQF3(BrvWaLdGVUIlqZt5vWXoRSI)IDTXL3E6SI3aSDBX9dE5x(BMWV5QgL2Jd(CJ79e37P47HJvQWe3dAXRVT5V6yfazyVTj7Yyp)mWg5aBKdSrxbM37)(CM6mZBkdDAZEmljyTxyeBBC1EhzIVGi5XSLlIJc2VOtDrCCu667tYYs(w6I)8RzdDY)RG(pDgRO8WPMBTc694zozkobH3FYNU(KiriIxHsTX34Zt1Crvn4XTtPKgfZnyblNZBygdEbcDt5HgPCxTMqw1GGa1rDnwXqjT)qNCGSkDrej7kVmibUYORLglq8YKByKRdXxYCS6(8ArxNIWEm4kzPOk4JgAHHjxqg3XYHHDOe9fHHbjNcxheoucZsQ8(HMTppAYiqVefxnw5vLANF(AtBTh29IcBTx7EhfLDiKjTf2hSFy7pMMI0hJQRoXNWEa5TxciZtJIIXoSJvyNMLlle2l6MYR5W0hyrpRPZNz6zAbErXWyWzD5W9WMYnQoDo3CAWUND51mzf)eIEGcng)XN8Tg43lVuTAvlxBpmOa(Kj48)fPuYHrQ)FyI77VMl2tPFNq3zbHoPDq)Co0dim9gee6pp
```

However, note that the load conditions on that aura specify that you must be a discipline priest for the aura to load. I suspect that you created an aura specifically for shadow priest, and then did not adjust the load conditions to allow for discipline priest too.

Thus, I am closing this ticket.

**Comment by Haxxen** [comment]
Hm. Your Aura isnt loading/showing for me either.

I'm made a awful lot auras for every spec. I mean i cant do custom code, but i know what load conditions are. I specifcally wrote, that if i just replace the spell in the CD-Trigger, the Aura will work with another spell. That wouldnt be possible if the load conditions were wrong. 

It can't be a translation error either, because spell id wont work too. 

Could it be, that another weakaura interfers? i have like 1,5k Auras (self built interface for 36 Specs) but that happened never to me.
![image](https://user-images.githubusercontent.com/60307067/73132663-4d66ca00-401e-11ea-881e-2c519f841d72.png)


This is your aura. Im Discipline speced. didnt change anything on your string. CDtrigger is set to show always. It shows the Symbol for "loaded" If i close the WA-Windows, the aura disappears. Load condition is priest and discipline. When shadowfiend gets on cooldown nothing happens. I tried it again without any other addons. Doesnt do it.

**Comment by mrbuds** [comment]
I tested rivers's aura and it doesn't show for me either, but checking "Exact Spell Match" fix the issue.

**Comment by InfusOnWoW** [comment]
Then that's probably blizzard's fault, I do have a good idea where the problem is.

The original reporter is using a german client, and I assume you are using French too?

**Comment by mrbuds** [comment]
I'm using english client

**Comment by InfusOnWoW** [comment]
That's strange, I'll try to figure out what's going on anyway.

**Comment by Haxxen** [comment]
2 little things:

 1. The issue is labeled classic. I didn't test that in classic.

2. Is it a good idea to use a clean fresh Install for testings things like this? I mean fresh WTF/Cache-Folder and no edited CVars? My WoW-Install is fairly old.

**Comment by InfusOnWoW** [comment]
With empty weakauras settings and without any other addons enabled is usually enough to ensure that there are no outside influences.

**Comment by InfusOnWoW** [comment]
I do remember a similar issue with spells for Warriors, that depended on a lot of details.

Specifally, it depended on the talent setup of all specs in regards to the Shadowfiend and which spec you logged in, and which spec you made the aura.

So can you create a step by step instruction that starts with a clean start of the game, and note down which spec you logged in, which talents you had and so on?

**Comment by Haxxen** [comment]
Ok i will do that later and post here.

**Comment by Haxxen** [comment]
-Startet game
-All Addons off except the weakaura ones
-Logged in as shadow
![image](https://user-images.githubusercontent.com/60307067/73506025-8adba680-43d4-11ea-9f60-ce0b4aab8533.png)
-Respeced to disc
![image](https://user-images.githubusercontent.com/60307067/73506045-a3e45780-43d4-11ea-92c0-ac45452c0d7f.png)
- Make a new Aura (as disc) : Use Template -> Symbol standard -> Abilities -> Shadowfiend --> Cooldowntracking ---> Close Weakaurawindow --> doesnt load
-change to exact spell match --> works 
-Change back --> doesn't work again
-Log out as Disc to characterscreen
-Log in as disc this time
**- ding ding ding --> aura loads without "exact spell match" ticked**
-switch do shadow and log out
-lock back in as shadow
-switch to disc --> aura wont load again without"exact spell match" ticked

everytime i log in as shadow and switch --> aura is gone. Then just log out and back in as disc --> aura works

I hope that helps.

**Comment by InfusOnWoW** [comment]
I can't reproduce that with your steps. Even with your talent setup and all other addons disabled.
I have no idea what else could affect this, as I'm also using the german client to try to reproduce it.
And I would assume that there are no azerite traits that affect the shadow fiend. 

It's clearly some kind of bug in blizzard's code for overriden spells.

I would assume that one of IsPlayerSpell, IsSpellKnown, GetSpellInfo returns wrong data after logging on as shadow and switching.

While I would prefer to get to the bottom of this, so that we can report it to Blizzard on the slight chance that they would fix it, I guess we have found a solution that works for you, and I don't feel like investing even more time into this.

That's not really satisfying, but at least we have helped you.
